server/activate.go

```diff
@@ -43,9 +46,9 @@ func (p *Plugin) canSendDiagnostics() bool {
 
 func (p *Plugin) ensureBotExists() *model.AppError {
```

Comment:
Why exactly this has to be `*model.AppError` and not `error`?

Reply:
I wanted to use `error` wherever possible, but I ran into some weird problems when the two types were mixed, so I've been changing back to `*model.AppError`. If I remember correctly, I had code like this where returning a nil error would somehow evaluate `err != nil` to true due to the type of nil being incorrect.
```go
func DoSomething() error {
    var appErr *model.AppError
    return appErr
}

if err := p.DoSomething(); err != nil {
    // This code would run
}
```

Reply:
Now that I think of it, it might be something to do with our overuse of pointers. I wonder if we ended up with something like `err` being a pointer to nil instead of a nil pointer itself.